src/java.base/share/classes/java/util/stream/AbstractPipeline.java

```diff
@@ -466,7 +466,18 @@ final StreamShape getSourceShape() {
 
     @Override
     final <P_IN> long exactOutputSizeIfKnown(Spliterator<P_IN> spliterator) {
-        return StreamOpFlag.SIZED.isKnown(getStreamAndOpFlags()) ? spliterator.getExactSizeIfKnown() : -1;
+        long size = StreamOpFlag.SIZED.isKnown(getStreamAndOpFlags()) ? spliterator.getExactSizeIfKnown() : -1;
```

Comment:
hey, 

small detail but maybe :

```
return StreamOpFlag.SIZED.isKnown(getStreamAndOpFlags()) ? adjustSize(spliterator.getExactSizeIfKnown()) : -1
```
this prevent local long `size` variable creation and obvious second ternary `size == -1` return -1

otherwise everything else good job 😃